ArmRobot/Assets/Scripts/RobotAgent.cs

```diff
@@ -40,46 +40,66 @@ public override void CollectObservations(VectorSensor sensor)
         float[] rotations = robotController.GetCurrentJointRotations();
         foreach (float rotation in rotations)
         {
-            // normalize rotation to [0, 1] range
-            float normalizedRotation = rotation / 360.0f;
+            // normalize rotation to [-1, 1] range
+            float normalizedRotation = (rotation / 360.0f) %  1f;
             sensor.AddObservation(normalizedRotation);
         }
 
+        foreach (var joint in robotController.joints)
```

Comment:
We don't really need to loops here, since under the covers, the other loop is just going through the joints as well. We could just use this new loop. We could add
```
ArticulationJointController jointController = joint.robotPart.GetComponent<ArticulationJointController>();
float rotation = jointController.CurrentPrimaryAxisRotation();
float normalizedRotation = (rotation / 360.0f) %  1f;
sensor.AddObservation(normalizedRotation);
```
to it, and then delete the loop above.